pam/gdm/conversation_test.go

```diff
@@ -99,3 +129,569 @@ func TestSendToGdm(t *testing.T) {
 		})
 	}
 }
+
+func TestDataSend(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(pam_test.MaybeDoLeakCheck)
+	testCases := map[string]struct {
+		value *Data
+
+		wantReturn []byte
+		wantError  error
+	}{
+		"nil return": {
+			value: &Data{
+				Type: DataType_event,
+				Event: &EventData{
+					Type: EventType_brokerSelected,
+					Data: &EventData_BrokerSelected{},
+				},
+			},
+
+			wantReturn: []byte("null"),
+		},
+
+		"version data": {
```

Comment:
Do you mind improving the test case names so they are more of a description of what's expected to happen? (a bit more similar to what you did for the error cases)